### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by Pritheesh**
Name     :   Pritheesh Panchmahalkar
SMU ID  :   47524741

### CODE REVIEW

File: ex01.exs [hunk 1]
```diff
@@ -26,8 +26,25 @@ defmodule Ex01 do
         2 is the program well laid out,  appropriately using indentation,
           blank lines, vertical alignment
```
Comment (pragdave) [inline ex01.exs:27]: :white_check_mark:

File: ex02.exs [hunk 2]
```diff
@@ -1,6 +1,27 @@
 
 defmodule Ex02 do
 
+  def new_counter(value \\ 0) do
+    { :ok, counter } = Agent.start_link( fn -> value end)
+    counter
+  end
+
+  def next_value(count) do
```
Comment (pragdave) [inline ex02.exs:9]: Nice layout below...

File: ex03.exs [hunk 3]
```diff
@@ -58,11 +58,53 @@ defmodule Ex03 do
         5 elegant use of language features or libraries
 
   """
-
+  # Implementation using Tasks
   def pmap(collection, process_count, function) do
-    « your code here »
+
```
Comment (pragdave) [inline ex03.exs:63]: this pipeline is a thing of beauty, no?

File: ex03.exs [hunk 4]
```diff
@@ -58,11 +58,53 @@ defmodule Ex03 do
...
+    div( Enum.count(collection)-1, process_count ) + 1
   end
 
+  # Helper function for spawnlink
+#  defp get_result(pid) do
+#    receive do
+#      { ^pid, result } -> result
+#    end
+#  end
+#
```
Comment (pragdave) [inline ex03.exs:100]: I had to think about this a little, to make sure that the `Enum.map` below was running in the correct process. 

This is why we like tasks :)

**Comment by pragdave** [comment]
Nice work. 💯

File: ex03.exs (hunk 3 above)
Comment (Pritheesh) [reply ex03.exs:63]: Yes, I'm loving it! :D

**Pull request closed without merging**